### ISSUE TITLE
Extracting an entire archive into a folder results in an Error

### ISSUE THREAD

**Issue by gregcotten** [abbeycode/UnzipKit#2]
Error Domain=UZKErrorDomain Code=104 "The operation couldn’t be completed. Error extracting files from the archive" UserInfo=0x618000477640 {NSLocalizedFailureReason=Error extracting files from the archive}

Thoughts?

**Comment by abbeycode** [comment]
Would you be able to send a link to the file you're having problems with? And does it open fine with other Unrar libraries/apps?

**Comment by gregcotten** [comment]
@abbeycode 

Here it is! https://github.com/ampas/aces-dev/archive/v1.0.zip

I ended up using NSTask with /usr/bin/ditto (ugh). Would love to replace it with a more elegant solution.

**Comment by abbeycode** [comment]
Thanks for that. I'll take a look as soon as I can. Did you try stepping into the code yet, to see what is happening inside UnzipKit?

**Comment by gregcotten** [comment]
I did. Sorry for the terse issue - wasn’t at my computer when I sent it. The error comes when I extract the entire archive to a destination. When it tries to write the NSData of the first file to the destination folder, the writeToURL:... writeSuccess is NO.

I think this is happening at [line 295 in UZKArchive.m](https://github.com/abbeycode/UnzipKit/blob/master/Source/UZKArchive.m#L295).

```
BOOL writeSuccess = [data writeToFile:extractPath
                              options:NSDataWritingAtomic
                                error:innerError];
if (!writeSuccess) {
    NSLog(@"Failed to extract file to path: %@", extractPath);
    [self assignError:innerError code:UZKErrorCodeOutputError];
    return;
}
```

**Comment by abbeycode** [comment]
Definitely looks like a bug. It's trying to write a directory out as a file, which is failing. That should be simple enough to patch up. Thanks for the report!

**Comment by abbeycode** [comment]
This should be fixed in v1.0.2, out now on CocoaPods. Let me know if it works for you.

**Comment by gregcotten** [comment]
@abbeycode Thanks! Looks like it should be good. I'll try it out.
